blas/gonum/level3double.go

```diff
@@ -172,7 +172,7 @@ func (Implementation) Dtrsm(s blas.Side, ul blas.Uplo, tA blas.Transpose, d blas
 			return
 		}
 		for i := 0; i < m; i++ {
-			btmp := b[i*lda : i*lda+n]
```

Comment:
Oops. How did this not show up before? Can we have an issue to add a test for it if it's not going to be added now.

Reply:
The `blas/gonum` tests have only `lda = ldb` for these two functions, tests in `mat` don't cover them and most tests in `lapack/gonum` that do also have `lda = ldb` except for `Dgetri` which eventually calls `Dtrsm` with `lda < ldb`. So it's not surprising that it didn't show up before although the `Dgetri` makes it a bit mysterious.

I'll see how much work would be to modify the tests to trigger failure without this fix and if not too much, I'll add it to this PR.

Speaking of Dgetri, I'll have to take a closer look at it. Its test uses a very crude tolerance but some quick experiments show that it's necessary only for matrix sizes > 64. Sizes 64 and smaller can bear tolerances in the identity matrix test as low as 1e-14. At 64 the algorithm switches to blocked code but the big jump in the accuracy of the result is disconcerting.